src/sentry/preprod/api/endpoints/project_preprod_artifact_download.py

```diff
@@ -0,0 +1,91 @@
+import posixpath
+
+from django.http.response import FileResponse, HttpResponseBase
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.endpoints.project_release_file_details import ClosesDependentFiles
+from sentry.models.files.file import File
+from sentry.preprod.authentication import LaunchpadRpcSignatureAuthentication
+from sentry.preprod.models import PreprodArtifact
+
+
+@region_silo_endpoint
+class ProjectPreprodArtifactDownloadEndpoint(ProjectEndpoint):
+    owner = ApiOwner.EMERGE_TOOLS
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
```

Comment:
We should avoid marking this public in the future, as public endpoints become part of the customer facing API docs.